Tasks/AzureRmWebAppDeploymentV4/operations/AzureAppServiceUtility.ts

```diff
@@ -253,17 +268,39 @@ export class AzureAppServiceUtility {
 
         if(!!releaseDefinitionId) {
             // Task is running in Release
-            let buildDefintionId = tl.getVariable("build.definitionId");
+            tl.debug("Artifact Source Alias is: "+ artifactAlias);
+            let artifactType = tl.getVariable(`release.artifacts.${artifactAlias}.type`);
+            let buildDefinitionUrl = "";
+            let buildDefintionId = "";
+
+            // Get build definition info only when artifact type is build.
+            if(artifactType && artifactType.toLowerCase() == "build") {
+                buildDefintionId = tl.getVariable("build.definitionId");
+                let buildProjectId = tl.getVariable("build.projectId") || projectId;
+
+                if(artifactAlias) {
```

Comment:
can it be null here?

Reply:
Yes see above comment.

Reply:
How can you find out the artifact type if the alias is null?